src/main/java/io/opentelemetry/contrib/generator/telemetry/cli/CLIProcessor.java

```diff
@@ -99,31 +106,36 @@ private static Options getOptions() {
 
     private static PayloadHandler getPayloadHandler(String targetEnvYAML) {
         TargetEnvironmentDetails targetEnvironmentDetails = getTargetEnvDetails(targetEnvYAML);
-        if (StringUtils.defaultString(targetEnvironmentDetails.getUsername()).isBlank()) {
-            throw new GeneratorException("Missing username in environment target YAML");
-        }
-        if (StringUtils.defaultString(targetEnvironmentDetails.getPassword()).isBlank()) {
-            throw new GeneratorException("Missing password in environment target YAML");
-        }
         String nonNullRestURL = StringUtils.defaultString(targetEnvironmentDetails.getRestURL());
         String nonNullGRPCHost = StringUtils.defaultString(targetEnvironmentDetails.getGRPCHost());
         String nonNullGRPCPort = StringUtils.defaultString(targetEnvironmentDetails.getGRPCPort());
         if (nonNullRestURL.isBlank() && (nonNullGRPCHost.isBlank() || nonNullGRPCPort.isBlank())) {
             throw new GeneratorException("Either restURL (for REST endpoint) or gRPCHost & gRPCPort (for gRPC endpoint) " +
                     "must be provided in environment target YAML");
         }
+        AuthHandler authHandler;
+        if (targetEnvironmentDetails.getAuthMode().equalsIgnoreCase("NONE")) {
```

Comment:
what happens if I do not set an authMode in the config file, will it default to "none" or to "basic"? or throw an error?

Reply:
An error will be thrown. As part of this [issue](https://github.com/cisco-open/test-telemetry-generator/issues/16) we will be supporting OAuth as well which will make things more confusing. Making this a mandatory arg seems cleaner.

Reply:
makes sense.